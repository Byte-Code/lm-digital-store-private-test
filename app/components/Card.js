import React, { Component, PropTypes } from 'react';
import styled from 'styled-components';

const CardBody = styled.div`
  display: ${props => props.isVisible};
`;

export default class Card extends Component {
  static propTypes = {
    TitleComponent: PropTypes.func.isRequired,
    children: PropTypes.node.isRequired,
  }

  constructor(props) {
    super(props);

    this.state = {
      expanded: false
    };
  }

  componentWillUpdate(prevProps) {
    if (prevProps !== this.props) {
      this.setState({ expanded: false });
    }
  }

  toggleCard() {
    const currentState = this.state.expanded;
    this.setState({ expanded: !currentState });
  }

  render() {
    const { TitleComponent, children } = this.props;
    const { expanded } = this.state;
    const isVisible = expanded ? 'block' : 'none';

    return (
      <div>
        <div onTouchTap={this.toggleCard.bind(this)}>
          <TitleComponent expanded={expanded} />
        </div>
        <CardBody
          isVisible={isVisible}
          className="animated fadeIn"
        >
          {children}
        </CardBody>
      </div>
    );
  }
}
