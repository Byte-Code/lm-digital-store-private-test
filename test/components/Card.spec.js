import React from 'react';
import { shallow } from 'enzyme';

import Card, { TitleWrapper } from '../../app/components/Card';

describe('Card', () => {
  const TitleComponent = () => (<div>Title</div>);
  const children = <div>Children</div>;
  const result = shallow(
    <Card TitleComponent={TitleComponent}>
      {children}
    </Card>
  );

  it('should initally be closed', () => {
    expect(result.state('expanded')).toBe(false);
    expect(result).toMatchSnapshot();
  });

  it('should open once the title is clicked', () => {
    result.find(TitleWrapper).simulate('click');
    expect(result.state('expanded')).toBe(true);
    expect(result).toMatchSnapshot();
  });

  it('should close if the title is clickedAgain', () => {
    result.find(TitleWrapper).simulate('click');
    expect(result.state('expanded')).toBe(false);
    expect(result).toMatchSnapshot();
  });
});
