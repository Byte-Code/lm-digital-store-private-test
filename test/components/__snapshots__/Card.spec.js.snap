// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should close if the title is clickedAgain 1`] = `
<div>
  <styled.div
    onClick={[Function]}
  >
    <TitleComponent
      expanded={false}
    />
  </styled.div>
  <styled.div
    className="animated fadeIn"
    isVisible="none"
  >
    <div>
      Children
    </div>
  </styled.div>
</div>
`;

exports[`Card should initally be closed 1`] = `
<div>
  <styled.div
    onClick={[Function]}
  >
    <TitleComponent
      expanded={false}
    />
  </styled.div>
  <styled.div
    className="animated fadeIn"
    isVisible="none"
  >
    <div>
      Children
    </div>
  </styled.div>
</div>
`;

exports[`Card should open once the title is clicked 1`] = `
<div>
  <styled.div
    onClick={[Function]}
  >
    <TitleComponent
      expanded={true}
    />
  </styled.div>
  <styled.div
    className="animated fadeIn"
    isVisible="block"
  >
    <div>
      Children
    </div>
  </styled.div>
</div>
`;
